index.js

```diff
@@ -1081,20 +1090,38 @@
           }
         }
 
-        // activates page tracking
-        if (trackRoutes) {
-          $rootScope.$on(pageEvent, function () {
-            // Apply $route based filtering if configured
-            if (readFromRoute) {
-              // Avoid tracking undefined routes, routes without template (e.g. redirect routes)
-              // and those explicitly marked as 'do not track'
-              if (!$route.current || !$route.current.templateUrl || $route.current.doNotTrack) {
+        // This section adds different listeners based on the configuration
+        var eventListener;
+        if (typeof trackRouteFn === 'function') {
+          eventListener = trackRouteFn;
+        } else if (trackRoutes && hybridMobileSupport) {
```

Comment:
The following else-if sections are all sub-branches off of `trackRoutes` being true. Do that check and then do an inner if-else block to handle the variants. e.g.,

```javascript
} else if (trackRoutes) {
   if (hybridMobileSupport) {
      ...
   } else if (!readFromRoute) {
      ...
   } else { // readFromRoute
      ...
   }
}
```

That will be more readable as to what is going on in the logic.